README.md

```diff
@@ -16,82 +20,811 @@ yarn add gotrue-js
-  error => console.log("Failed to delete user: %o", error)
-)
+import GoTrue from "gotrue-js";
+
+// Instantiate the GoTrue auth client with an optional configuration
+
+auth = new GoTrue({
+  APIUrl: "https://<your domain name>/.netlify/identity",
+  audience: "",
+  setCookie: false
+});
+```
+
+### GoTrue configuration
+
+APIUrl: The absolute path of the GoTrue endpoint. To find the `APIUrl`, go to `Identity` page of your Netlify site dashboard.
+
+audience(optional): `audience` is one of the pre-defined [JWT payload](https://tools.ietf.org/html/rfc7519#section-4.1.3) claims. It's an optional attribute which is set to be empty by default. If you were hosting your own identity service and wanted to support [multitenancy](https://en.wikipedia.org/wiki/Multitenancy), you would need `audience` to separate the users.
+
+setCookie(optional): set to be `false` by default. If you wish to use the `remember me` functionality, set the value to be `true`.
+
+### Error handling
+
+If an error occurs during the request, the promise may be rejected with an Error, `HTTPError`, `TextHTTPError`, or `JSONHTTPError`. See [micro-api-client-lib error types](https://github.com/netlify/micro-api-client-lib#class-httperror-extends-error).
+
+## Authentication methods
+
+### Create a new user
+
+Create a new user with the specified email and password
+
+```js
+auth.signup(email, password);
+```
+
+Example usage:
+
+```js
+auth
+  .signup(email, password)
+  .then(response => console.log("Confirmation email sent", response))
+  .catch(error => console.log("It's an error", error));
+```
+
+Example response object:
+
+```js
+{
+  id: 'example-id',
+  aud: '',
+  role: '',
+  email: 'example@example.com',
+  confirmation_sent_at: '2018-04-27T22:36:59.636416916Z',
+  app_metadata: { provider: 'email' },
+  user_metadata: null,
+  created_at: '2018-04-27T22:36:59.632133283Z',
+  updated_at: '2018-04-27T22:37:00.061039863Z'
+}
+```
+
+Also, make sure the `Registration preferences` under `Identity settings` in your Netlify dashboard are set to `Open`.
+
+[registration preferences](https://github.com/netlify/gotrue-js/blob/update-readme/src/images/identity-settings-registration.png)
+
+If the registration preferences is set to be `Invite only`, you'll get an error message like this:
+`{code: 403, msg: 'Signups not allowed for this instance'}`
+
+### Confirm a new user signup
+
+This function confirms a user sign up via a unique confirmation token
+
+```js
+auth.confirm(token);
+```
+
+When a new user signed up, a confirmation email will be sent to the user if `Autoconfirm` isn't turned on under the [identity settings](https://www.netlify.com/docs/identity/#adding-users).
+
+In the email, there's a link that says "Confirm your email address".
+When a user clicks on the link, it'll be redirected to the site with a [fragment identifier](https://en.wikipedia.org/wiki/Fragment_identifier) `#confirmation_token=Iyo9xHvsGVbW-9A9v4sDmQ` in the URL.
```

Comment:
Hidden from the browser? Or hidden from the email client?

Reply:
So the link in the mail is an encoded URL, when ppl click on it, it'll be opened in the browser, and the redirect will hide the token in the URL bar. Let me know if it makes sense.